src/arithmetic.jl

```diff
@@ -731,8 +744,25 @@ end
     return nothing
 end
 
-div!(v::Taylor1, b::NumberNotSeries, a::Taylor1, k::Int) =
-    div!(v::Taylor1, Taylor1(b, a.order), a, k)
+@inline function div!(c::Taylor1, a::NumberNotSeries,
+        b::Taylor1, k::Int)
```

Comment:
Perhaps we should impose `c::Taylor1{T}` as well as `b::Taylor1{T}`...

Reply:
I agree, although I think we can make an extra tweak to avoid some allocations for `T<:TaylorN{<:NumberNotSeries}`, so I added an extra method for the latter case (will fix it in the next few commits)